ccl/serverccl: TestServerControllerMultiNodeTenantStartup failed

ccl/serverccl.TestServerControllerMultiNodeTenantStartup [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11500671?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11500671?buildTab=artifacts#/) on master @ [b7d514b5d84a6cb298fd84dc5e0b84bbe6f681f9](https://github.com/cockroachdb/cockroach/commits/b7d514b5d84a6cb298fd84dc5e0b84bbe6f681f9):


```
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:416: condition failed to evaluate within 45s: pq: service unavailable for target tenant (hello)
    server_controller_test.go:402: stopping test cluster
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/c101b7a464a1afc1f5af0cd85792187e/logTestServerControllerMultiNodeTenantStartup2138478687
--- FAIL: TestServerControllerMultiNodeTenantStartup (101.11s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/multi-tenant @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestServerControllerMultiNodeTenantStartup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30981

ccl/serverccl.TestServerControllerMultiNodeTenantStartup [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11549383?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11549383?buildTab=artifacts#/) on master @ [36bd8296bad475a8183e474bf4aeb08bb83f8a2d](https://github.com/cockroachdb/cockroach/commits/36bd8296bad475a8183e474bf4aeb08bb83f8a2d):


```
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:416: condition failed to evaluate within 45s: pq: service unavailable for target tenant (hello)
    server_controller_test.go:402: stopping test cluster
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/c101b7a464a1afc1f5af0cd85792187e/logTestServerControllerMultiNodeTenantStartup3038360093
--- FAIL: TestServerControllerMultiNodeTenantStartup (106.92s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestServerControllerMultiNodeTenantStartup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/serverccl.TestServerControllerMultiNodeTenantStartup [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11598459?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11598459?buildTab=artifacts#/) on master @ [b994d025c678f495cb8b93044e35a8c59595bd78](https://github.com/cockroachdb/cockroach/commits/b994d025c678f495cb8b93044e35a8c59595bd78):


```
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:424: connection not ready (will retry): pq: service unavailable for target tenant (hello)
    server_controller_test.go:416: condition failed to evaluate within 45s: pq: service unavailable for target tenant (hello)
    server_controller_test.go:402: stopping test cluster
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/c101b7a464a1afc1f5af0cd85792187e/logTestServerControllerMultiNodeTenantStartup3664234031
--- FAIL: TestServerControllerMultiNodeTenantStartup (103.92s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestServerControllerMultiNodeTenantStartup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/serverccl/server_controller_test.go
+++ b/pkg/ccl/serverccl/server_controller_test.go
@@ -388,7 +388,7 @@ func TestServerControllerMultiNodeTenantStartup(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
-
+	skip.UnderDeadlock(t, "slow under deadlock")
 	t.Logf("starting test cluster")
 	numNodes := 3
 	tc := serverutils.StartCluster(t, numNodes, base.TestClusterArgs{

